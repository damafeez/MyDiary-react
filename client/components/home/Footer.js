/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';
import './Footer.scss';
import icons from '../../assets/icons.svg';

export default ({
  handleAdd,
  showAdd,
  updateMode,
  entryIsSelected,
  handleDelete,
  handleUpdate,
  history,
}) => (
  <footer>
    <button onClick={() => history.push('/profile')} type="button" title="Settings" className="settings drop-container">
      <svg className="icon">
        <use xlinkHref={`${icons}#icon-settings`} />
      </svg>
      {/* <div className="drop">
            <div className="check">
              <input type="checkbox" checked id="show-notification" />
              <label htmlFor="show-notification">Reminder</label>
            </div>
            <p id="info" />
            <p onClick="logout()" className="btn round">Logout</p>
          </div> */}
    </button>
    <button type="button" title="New entry" onClick={handleAdd} className={`add ${showAdd && !updateMode ? 'active' : ''}`}>
      <svg className="icon">
        <use xlinkHref={`${icons}#icon-plus`} />
      </svg>
    </button>
    {entryIsSelected && (
      <button type="button" title="Edit Entry" onClick={handleUpdate} className="edit">
        <svg className="icon">
          <use xlinkHref={`${icons}#icon-edit-3`} />
        </svg>
      </button>
    )}
    {entryIsSelected && (
      <button type="button" title="Delete Entry" onClick={handleDelete} className="delete">
        <svg className="icon">
          <use xlinkHref={`${icons}#icon-trash-2`} />
        </svg>
      </button>
    )}
  </footer>
);
