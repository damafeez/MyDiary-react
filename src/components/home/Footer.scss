
footer {
  bottom: 0;
  justify-content: space-between;
  & > button {
    margin: 0 2rem;
    background-color: transparent;
    &, & .icon {
      transition: transform .3s ease-in;
    }
    &:hover {
      transform: translateY(-3px)
    }
  }
  .settings {
    z-index: 5;
    display: none;
    &::after {
      border-color: transparent transparent #fcfafa #fcfafa;
    }
    .drop {
      width: 15rem;
      background: #fcfafa;
      display: inline-flex;
      flex-direction: column;
      align-items: center;
      font-size: 1.5rem;
      &  > * {
        margin-bottom: 1rem;
        &.check {
          border-bottom: 1px solid #f1f1f1;
          display: flex;
          width: 100%;
          align-items: center;
          justify-content: space-between;
          padding: 1rem 0;
          margin-bottom: 2rem;
        }
        &#info {
          color: gray;
        }
      }
    }
  }
  .add {
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    background-color: #43a047;
    display: flex;
    justify-content: center;
    align-items: center;
    fill: white;
    box-shadow: 0 16px 26px -10px rgba(60, 72, 88, 0.56),
    0 4px 25px 0px rgba(0, 0, 0, 0.12),
    0 8px 10px -5px rgba(60, 72, 88, 0.2);
    /* animation-name: pulse; */
    animation-iteration-count: 10;
    animation-duration: 2s;
    &.active .icon {
      transform: rotate(45deg);
    }
    &::after {
      border-color: transparent transparent #43a047 #43a047;
    }
    .drop {
      background: #43a047;
      color: white;
    }
  }
  .delete {
    fill: #ff5925;
  }
  .edit {
    order: -1;
  }
  @media screen and (min-width: 550px) {
    justify-content: flex-start;
    .edit {
      margin-left: auto;
      order: 0;
    }
    .settings {
      display: initial;
    }
  }
}