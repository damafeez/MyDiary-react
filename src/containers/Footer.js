import React, { Component } from 'react';
import { connect } from 'react-redux';
import FooterUI from '../components/home/Footer';
import scrollAction from '../actions/scroll';

// eslint-disable-next-line react/prefer-stateless-function
export class Footer extends Component {
  render() {
    const { scroll, showAdd } = this.props;
    return <FooterUI showAdd={showAdd} scroll={scroll} />;
  }
}
const mapStateToProp = (state) => ({ showAdd: state.scroller.showAdd });

export default connect(mapStateToProp, { scroll: scrollAction })(Footer);
