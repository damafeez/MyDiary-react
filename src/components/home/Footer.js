/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React, { Component } from 'react';
import './Footer.scss';
import icons from '../../assets/icons.svg';

export default class AddEntry extends Component {
  state = {
    title: '',
    body: '',
  }

  render() {
    const { title, body } = this.state;
    return (
      <footer>
        <button type="button" title="Settings" className="settings drop-container">
          <svg className="icon">
            <use xlinkHref={`${icons}#icon-settings`} />
          </svg>
          {/* <div className="drop">
            <div className="check">
              <input type="checkbox" checked id="show-notification" />
              <label htmlFor="show-notification">Reminder</label>
            </div>
            <p id="info" />
            <p onClick="logout()" className="btn round">Logout</p>
          </div> */}
        </button>
        <button type="button" title="New entry" onClick="addDiary()" className="add">
          <svg className="icon">
            <use xlinkHref={`${icons}#icon-plus`} />
          </svg>
        </button>
        <button type="button" title="Edit Entry" onClick="editDiary()" className="edit">
          <svg className="icon">
            <use xlinkHref={`${icons}#icon-edit-3`} />
          </svg>
        </button>
        <button type="button" title="Delete Entry" onClick="confirmAction('Are you sure you want to delete?', deleteEntry)" className="delete">
          <svg className="icon">
            <use xlinkHref={`${icons}#icon-trash-2`} />
          </svg>
        </button>
      </footer>
    );
  }
}
