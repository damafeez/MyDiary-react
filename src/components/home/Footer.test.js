import React from 'react';
import Footer from './Footer';

const setup = (overrideProps) => {
  const props = {
    entryIsSelected: true,
    updateMode: false,
    showAdd: true,
    handleAdd: jest.fn(),
    history: { push: jest.fn() },
    entries: [{ title: 'this title', body: 'this body' }],
  };
  const wrapper = shallow(<Footer {...props} {...overrideProps} />);
  return { props, wrapper };
};

it('matches snapshot', () => {
  const { wrapper } = setup({ currentIndex: 1 });
  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot', () => {
  const { wrapper } = setup({ currentIndex: 1, showAdd: false });
  wrapper.find('button.settings').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
