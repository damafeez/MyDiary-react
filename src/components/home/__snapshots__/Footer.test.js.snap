// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    currentIndex={1}
    entries={
      Array [
        Object {
          "body": "this body",
          "title": "this title",
        },
      ]
    }
    entryIsSelected={true}
    handleAdd={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    showAdd={true}
    updateMode={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="settings drop-container"
          onClick={[Function]}
          title="Settings"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-settings"
            />
          </svg>
        </button>,
        <button
          className="add active"
          onClick={[MockFunction]}
          title="New entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-plus"
            />
          </svg>
        </button>,
        <button
          className="edit"
          title="Edit Entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-edit-3"
            />
          </svg>
        </button>,
        <button
          className="delete"
          title="Delete Entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-trash-2"
            />
          </svg>
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-settings"
            />
          </svg>,
          "className": "settings drop-container",
          "onClick": [Function],
          "title": "Settings",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-settings"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-settings",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-plus"
            />
          </svg>,
          "className": "add active",
          "onClick": [MockFunction],
          "title": "New entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-plus"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-plus",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-edit-3"
            />
          </svg>,
          "className": "edit",
          "onClick": undefined,
          "title": "Edit Entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-edit-3"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-edit-3",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-trash-2"
            />
          </svg>,
          "className": "delete",
          "onClick": undefined,
          "title": "Delete Entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-trash-2"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-trash-2",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="settings drop-container"
            onClick={[Function]}
            title="Settings"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-settings"
              />
            </svg>
          </button>,
          <button
            className="add active"
            onClick={[MockFunction]}
            title="New entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-plus"
              />
            </svg>
          </button>,
          <button
            className="edit"
            title="Edit Entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-edit-3"
              />
            </svg>
          </button>,
          <button
            className="delete"
            title="Delete Entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-trash-2"
              />
            </svg>
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-settings"
              />
            </svg>,
            "className": "settings drop-container",
            "onClick": [Function],
            "title": "Settings",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-settings"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-settings",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-plus"
              />
            </svg>,
            "className": "add active",
            "onClick": [MockFunction],
            "title": "New entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-plus"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-plus",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-edit-3"
              />
            </svg>,
            "className": "edit",
            "onClick": undefined,
            "title": "Edit Entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-edit-3"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-edit-3",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-trash-2"
              />
            </svg>,
            "className": "delete",
            "onClick": undefined,
            "title": "Delete Entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-trash-2"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-trash-2",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    currentIndex={1}
    entries={
      Array [
        Object {
          "body": "this body",
          "title": "this title",
        },
      ]
    }
    entryIsSelected={true}
    handleAdd={[MockFunction]}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/profile",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    showAdd={false}
    updateMode={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="settings drop-container"
          onClick={[Function]}
          title="Settings"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-settings"
            />
          </svg>
        </button>,
        <button
          className="add "
          onClick={[MockFunction]}
          title="New entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-plus"
            />
          </svg>
        </button>,
        <button
          className="edit"
          title="Edit Entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-edit-3"
            />
          </svg>
        </button>,
        <button
          className="delete"
          title="Delete Entry"
          type="button"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-trash-2"
            />
          </svg>
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-settings"
            />
          </svg>,
          "className": "settings drop-container",
          "onClick": [Function],
          "title": "Settings",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-settings"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-settings",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-plus"
            />
          </svg>,
          "className": "add ",
          "onClick": [MockFunction],
          "title": "New entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-plus"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-plus",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-edit-3"
            />
          </svg>,
          "className": "edit",
          "onClick": undefined,
          "title": "Edit Entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-edit-3"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-edit-3",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#icon-trash-2"
            />
          </svg>,
          "className": "delete",
          "onClick": undefined,
          "title": "Delete Entry",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#icon-trash-2"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#icon-trash-2",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "button",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="settings drop-container"
            onClick={[Function]}
            title="Settings"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-settings"
              />
            </svg>
          </button>,
          <button
            className="add "
            onClick={[MockFunction]}
            title="New entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-plus"
              />
            </svg>
          </button>,
          <button
            className="edit"
            title="Edit Entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-edit-3"
              />
            </svg>
          </button>,
          <button
            className="delete"
            title="Delete Entry"
            type="button"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-trash-2"
              />
            </svg>
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-settings"
              />
            </svg>,
            "className": "settings drop-container",
            "onClick": [Function],
            "title": "Settings",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-settings"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-settings",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-plus"
              />
            </svg>,
            "className": "add ",
            "onClick": [MockFunction],
            "title": "New entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-plus"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-plus",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-edit-3"
              />
            </svg>,
            "className": "edit",
            "onClick": undefined,
            "title": "Edit Entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-edit-3"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-edit-3",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#icon-trash-2"
              />
            </svg>,
            "className": "delete",
            "onClick": undefined,
            "title": "Delete Entry",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#icon-trash-2"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#icon-trash-2",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "button",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
